services/explorer-indexer/src/modules/dag/dag.service.ts

```diff
@@ -41,35 +43,67 @@ export default class DagService {
       transactionCount,
       transactions,
     } = { ...dagData };
-    const cherryPick: IDAG = {
-      hash: zeroX(hash),
-      pivot: zeroX(pivot),
-      tips,
-      level: parseInt(`${level}`, 16) || 0,
-      pbftPeriod: period,
-      timestamp: parseInt(timestamp, 16) || 0,
-      author: zeroX(author),
-      signature: zeroX(signature),
-      vdf: parseInt(vdf?.difficulty, 16) || 0,
-      transactionCount,
+
+    const txes = await findTransactionsByHashesOrFill(
       transactions,
-    };
-    const saved = await this.dagRepository.save(cherryPick);
-    if (saved) {
-      this.logger.log(`Registered new DAG ${saved.hash}`);
+      this.txRepository,
+      this.logger
+    );
+    let newDag = await this.dagRepository.findOneBy({ hash: zeroX(hash) });
```

Comment:
This can be new or existing so might be good to rename it. Is dag used anywhere? We could use just dag or currentDag. you decide.